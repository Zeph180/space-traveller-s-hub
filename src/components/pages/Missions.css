.missions-cont {
  margin-top: var(--tenPx);
  background-color: white;
}

.missions-cont > :nth-child(even) {
  background-color: lightgray;
}

.mission-header {
  box-sizing: border-box;
  width: 15.3%;
}

.desc-header {
  font-size: 16px;
  font-weight: bold;
  box-sizing: border-box;
  width: 60%;
  margin: var(--zero);
  padding: var(--five);
}

.status-desc {
  width: 10.85%;
  margin: var(--zero);
  padding: var(--five);
}

.mission {
  display: flex;
  flex-direction: row;
  width: 80%;
  margin-left: 10%;
  border: 1px solid lightgray;
}

.mission > * {
  border-right: 1px solid gray;
}

h3 {
  width: 15%;
  margin: var(--zero);
  padding-left: var(--five);
}

.mission-desc {
  width: 60%;
  margin: var(--zero);
  padding: var(--five);
  font-size: 13px;
  line-height: 18px;
}

.members-cont {
  width: 12%;
}

.buttons-cont {
  width: 13%;
}

.buttons-cont,
.members-cont {
  display: flex;
  justify-content: center;
  align-items: center;
}

.join-btn,
.leave-btn {
  padding: 5px;
  font-size: 16px;
  border-radius: 5px;
  cursor: pointer;
  background-color: transparent;
}

.leave-btn {
  color: red;
  border-color: red;
}

.member-badge,
.not-member-badge {
  color: var(--white);
  padding: var(--zero) 3px;
  border-radius: var(--five);
}

.member-badge {
  background-color: var(--blue);
}

.not-member-badge {
  background-color: gray;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
}
