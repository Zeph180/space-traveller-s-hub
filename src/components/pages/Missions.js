/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/prop-types */
import { useEffect } from 'react';
import './Missions.css';
import { useSelector, useDispatch } from 'react-redux';
import {
  getMissionsAsync,
  missionsActions,
} from '../redux/features/missions/missionsSlice';

const Missions = () => {
  const { missions } = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMissionsAsync());
  }, [dispatch]);

  return (
    <section className="missions-cont">
      <article className="mission">
        <h3 className="mission-header">Mission</h3>
        <p className="desc-header">Description</p>
        <p className="status-desc">Status</p>
      </article>
      {missions.map((mission) => (
        <>
          <Mission mission={mission} />
        </>
      ))}
    </section>
  );
};

const Mission = ({ mission }) => {
  const dispatch = useDispatch();

  const handleJoinMission = (e) => {
    const { id } = e.target;
    dispatch(missionsActions.joinMission(id));
  };

  const handleLeaveMission = (e) => {
    const { id } = e.target;
    dispatch(missionsActions.leaveMission(id));
  };

  return (
    <article className="mission">
      <h3>{mission.mission_name}</h3>
      <p className="mission-desc">{mission.description}</p>
      <div className="members-cont">
        {mission.joined === true ? (
          <span className="member-badge">Active member</span>
        ) : (
          <span className="not-member-badge">NOT A MEMBER</span>
        )}
      </div>
      <div className="buttons-cont">
        {mission.joined === true ? (
          <button
            onClick={handleLeaveMission}
            id={mission.mission_id}
            type="button"
            className="leave-btn"
          >
            Leave Mission
          </button>
        ) : (
          <button
            onClick={handleJoinMission}
            id={mission.mission_id}
            type="button"
            className="join-btn"
          >
            Join Mission
          </button>
        )}
      </div>
    </article>
  );
};

export default Missions;
