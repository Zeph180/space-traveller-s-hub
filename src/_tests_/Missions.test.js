import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../components/redux/app/Store';
import Missions from '../components/pages/Missions';

describe('Missions component', () => {
  test('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
