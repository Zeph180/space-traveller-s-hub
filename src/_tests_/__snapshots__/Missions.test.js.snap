// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component renders correctly 1`] = `
<section
  className="missions-cont"
>
  <article
    className="mission"
  >
    <h3
      className="mission-header"
    >
      Mission
    </h3>
    <p
      className="desc-header"
    >
      Description
    </p>
    <p
      className="status-desc"
    >
      Status
    </p>
  </article>
</section>
`;

exports[`Missions component should match the snapshot 1`] = `
<section
  className="mission-sec"
>
  <p>
    Missions
  </p>
  <div>
    <h3
      className="mission-name"
    >
      Mission 1
    </h3>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <p
              className="description"
            >
              Description 1
            </p>
          </td>
          <td>
            <span
              className="no-member"
            >
              Not a member
            </span>
          </td>
          <td>
            <button
              className="join-btn"
              id="mission1"
              onClick={[Function]}
              type="button"
            >
              Join Mission
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div>
    <h3
      className="mission-name"
    >
      Mission 2
    </h3>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <p
              className="description"
            >
              Description 2
            </p>
          </td>
          <td>
            <span
              className="member"
            >
              Active member
            </span>
          </td>
          <td>
            <button
              className="leave-btn"
              id="mission2"
              onClick={[Function]}
              type="button"
            >
              Leave Mission
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</section>
`;
